src/kirby/components/dropdown/dropdown.component.ts

```diff
@@ -0,0 +1,256 @@
+  }
+
+  scrollItemIntoView(index: number) {
+    const kirbyItems = this.kirbyItemsSlotted.length
+      ? this.kirbyItemsSlotted
+      : this.kirbyItemsDefault;
+    if (kirbyItems.length) {
+      const selectedKirbyItem = kirbyItems.toArray()[index];
+      if (selectedKirbyItem && selectedKirbyItem.nativeElement) {
+        const itemElement = selectedKirbyItem.nativeElement;
+        const scrollContainer = this.cardElement.nativeElement;
+        const itemTop = itemElement.offsetTop;
+        const itemBottom = itemElement.offsetTop + itemElement.offsetHeight;
+        const containerVisibleTop = scrollContainer.scrollTop;
+        const containerVisibleBottom = scrollContainer.clientHeight + scrollContainer.scrollTop;
+        if (itemTop < containerVisibleTop) {
+          scrollContainer.scrollTop = itemTop;
+        } else if (itemBottom > containerVisibleBottom) {
+          scrollContainer.scrollTop = itemBottom - scrollContainer.clientHeight;
+        }
+      }
+    }
+  }
+
+  @HostListener('keydown.tab', ['$event'])
+  private _onTab(event: KeyboardEvent) {
+    if (this.isOpen) {
+      event.preventDefault();
+      this.close();
+    }
+  }
+
+  @HostListener('keydown.enter')
+  @HostListener('keydown.escape')
+  @HostListener('blur')
+  private _onBlur() {
+    if (this.isOpen) {
+      this.close();
+    }
+  }
+
+  @HostListener('keydown.space', ['$event'])
+  private _onSpace(event: KeyboardEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+    if (!this.isOpen) {
+      this.open();
+    }
+  }
+
+  @HostListener('keydown.enter', ['$event'])
+  private _onEnter(event: KeyboardEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+    this.toggle();
+  }
+
+  @HostListener('keydown.arrowup', ['$event'])
+  @HostListener('keydown.arrowdown', ['$event'])
+  @HostListener('keydown.arrowleft', ['$event'])
+  @HostListener('keydown.arrowright', ['$event'])
+  private _onArrowKeys(event: KeyboardEvent) {
+    if (this.isOpen && (event.key === 'ArrowLeft' || event.key === 'ArrowRight')) {
+      // Mirror default HTML5 select behaviour - prevent left/right arrows when open:
+      return;
+    }
+    event.preventDefault();
+    let newIndex = -1;
+    if (event.key === 'ArrowUp' || event.key === 'ArrowLeft') {
+      // Select previous item:
+      newIndex = this.selectedIndex - 1;
+    }
+    if (event.key === 'ArrowDown' || event.key === 'ArrowRight') {
+      if (this.selectedIndex === undefined) {
+        // None selected, select first item:
+        newIndex = 0;
+      } else if (this.selectedIndex < this.items.length - 1) {
+        // Select next item:
+        newIndex = this.selectedIndex + 1;
+      }
+    }
+    if (newIndex > -1) {
+      this.selectItem(newIndex);
+    }
+  }
+
+  @HostListener('keydown.home', ['$event'])
+  @HostListener('keydown.end', ['$event'])
+  private _onHomeEndKeys(event: KeyboardEvent) {
```

Comment:
It's not possible to do AOT-compile `@HostBindings` to private functions - which is a requisite for producing an transpiled version of kirby. Instead - consider making the function public (but keep the underscore prefix to indicate that it should be consider `private`)?